packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -0,0 +1,118 @@
+/* Babel 7 transform */
+
+const babylon = require("babylon");
+const generate = require("@babel/generator").default;
+
+function printNode(ast) {
+  return generate(ast, {}).code;
+}
+
+function parse(code) {
+  return babylon.parse(code, {
+    allowImportExportEverywhere: true,
+    allowReturnOutsideFunction: true,
+    allowSuperOutsideMethod: true,
+  }).program;
+}
+
+const transform = require("nodent-transform").transform;
+
+function transformAsyncToPromises(api, options) {
+  let requiresTranspilation;
+  const runtime = options.runtime
+    ? typeof options.runtime === "string" ? options.runtime : "$nodent_runtime"
+    : null;
+  const opts = {
+    // Code generation options
+    es6target: false,
+    babelTree: true,
+    engine: false,
+    generators: false,
+    promises: true,
+    lazyThenables: false,
+    wrapAwait: true,
+    noRuntime: !runtime,
+    $runtime: runtime,
+    generatedSymbolPrefix: "$",
+    $return: "$return",
+    $error: "$error",
+    $arguments: "$args",
+    $Promise: "Promise",
+    $asyncspawn: "$asyncspawn",
+    $asyncbind: "$asyncbind",
+    $makeThenable: "$makeThenable",
+  };
+
+  if (options.codeGenerationOptions) {
+    const keys = Object.keys(options.codeGenerationOptions);
+    for (let i = 0; i < keys.length; i++) {
+      if (keys[i] in opts) {
+        opts[keys[i]] = options.codeGenerationOptions[keys[i]];
+      }
+    }
+  }
+
+  return {
+    visitor: {
+      Program: {
+        enter: function() {
+          requiresTranspilation = false;
+        },
+        exit: function(path, state) {
+          // Check if there was an async or await keyword before bothering to process the AST
+          if (!requiresTranspilation) return;
+
+          const newAst = transform(
+            {
+              // Input: the ast and filename
+              filename: state.filename,
+              ast: path.node,
+            },
+            opts,
+            {
+              // Helpers for the transformer:
+              // Parse a JS fragment into an AST
+              parse: parse,
+              // Print a node as JS source
+              printNode: printNode,
+              // Log a warning, eg: console.log.bind(console)
+              logger: false,
```

Comment:
It seems like this line throws together with nodent-transform as it just uses the value here as a function.

I am seeing this stack trace:

```
      at node_modules/nodent-transform/arboriculture.js:2327:25
          at Array.forEach (<anonymous>)
      at node_modules/nodent-transform/arboriculture.js:2326:50
      at treeWalk (node_modules/nodent-transform/arboriculture.js:2808:2)
      at down (node_modules/nodent-transform/arboriculture.js:2804:4)
      at node_modules/nodent-transform/arboriculture.js:2772:5
          at Array.forEach (<anonymous>)
      at forEachNodeKey (node_modules/nodent-transform/arboriculture.js:2767:8)
      at descend (node_modules/nodent-transform/arboriculture.js:2786:3)
      at node_modules/nodent-transform/arboriculture.js:2381:13
      at treeWalk (node_modules/nodent-transform/arboriculture.js:2808:2)
      at down (node_modules/nodent-transform/arboriculture.js:2804:4)
      at node_modules/nodent-transform/arboriculture.js:2769:44
          at Array.forEach (<anonymous>)
      at node_modules/nodent-transform/arboriculture.js:2769:15
          at Array.forEach (<anonymous>)
      at forEachNodeKey (node_modules/nodent-transform/arboriculture.js:2767:8)
      at descend (node_modules/nodent-transform/arboriculture.js:2786:3)
      at node_modules/nodent-transform/arboriculture.js:2381:13
      at treeWalk (node_modules/nodent-transform/arboriculture.js:2808:2)
      at hoistDeclarations (node_modules/nodent-transform/arboriculture.js:2213:9)
      at asyncTransforms (node_modules/nodent-transform/arboriculture.js:441:9)
      at asynchronize (node_modules/nodent-transform/arboriculture.js:422:9)
      at transform (node_modules/nodent-transform/arboriculture.js:2881:14)
      at PluginPass.exit (src/transformAsyncToPromises.js:67:26)
      at newFn (node_modules/@babel/traverse/lib/visitors.js:243:21)
```

Using an empty function helps. But I am not sure whether we should just ignore all the warnings.

Reply:
Fixed in nodent-transform@3.2.3 - `false` now disables reporting of warnings